technical_architecture/notification/add.rst

```diff
@@ -67,48 +61,113 @@ Create a Notification Factory
+
+    The optional route parameter will be:
 
-    $route = 'pim_importexport_export_execution_show';
-    $routeParams = ['id' => $jobExecutionId];
+    .. code-block:: php
 
-If the messageParams option is provided, it will be passed to the message when translating it.
+        $route = 'pim_importexport_export_execution_show';
+        $routeParams = ['id' => $jobExecutionId];
 
-The context allows to store some extra data in the notification, it is not displayed in the UI by default.
+6) Context (``array``)
+    The context allows to store some extra data in the notification, it is not displayed in the UI by default. Some important extra data you may use:
+        - ``actionType`` (``string``): this will be used to guess the icon to display on the notification
+        - ``showReportButton`` (``bool``): to hide/display the "report" label on the notification
 
-Now register your new factory with the proper tag:
+.. note::
+    You can see available notification icons on the `styleguide website <https://docs.akeneo.com/2.0/design_pim/styleguide/index.php#Templates-AknNotification>`_.
+
+For example, the ``create`` method of the NotificationFactory for mass edit notifications looks like that:
+
+.. code-block:: php
+
+    public function create($jobExecution)
+    {
+        $notification = new Notification();
+        $type = $jobExecution->getJobInstance()->getType();
+        $status = $this->getJobStatus($jobExecution);
+
+        $notification
+            ->setType($status)
+            ->setMessage(sprintf('pim_mass_edit.notification.%s.%s', $type, $status))
+            ->setMessageParams(['%label%' => $jobExecution->getJobInstance()->getLabel()])
+            ->setRoute('pim_enrich_job_tracker_show')
+            ->setRouteParams(['id' => $jobExecution->getId()])
+            ->setContext(['actionType' => $type]);
+
+        return $notification;
+    }
+
+Well, now we created our very own Notification Factory, **we need to register it** with the proper tag:
 
 .. code-block:: yaml
 
     services:
         acme_custom.notification.factory.custom_notification_factory:
             class: 'Acme\CustomBundle\Notification\CustomNotificationFactory'
             arguments:
-                - ['my_custom_name']
+                - ['my_custom_notification_name']
             tags:
                 - { name: pim_notification.factory.notification }
 
-With this tag you will be able to get your new factory from the dedicated registry.
+With this tag we will be able to get our new factory from the dedicated registry.
 
 Inject Services
 ---------------
 
-The notifier service is called: ``@pim_notification.notifier`` and the factory registry is
-``@pim_notification.registry.factory.notification``.
+Now that our Notification Factory is created and registered, we can build our own notifications!
+
+.. warning::
+    The Notification Factory we just created is **not responsible of sending notifications**, only to build them.
```

Comment:
`not responsible of sending notifications` => `not responsible for sending notifications`